### PR TITLE
Refactor CompanyProfileService

### PR DESCRIPTION
**Pull request by AlisonGriffiths**
Brought code inline with other services
Added unit test coverage

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/services/companyProfileService.ts [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
 
-    if (sdkResponse.httpStatusCode >= 400) {
-        throw logger.info(`${getCompanyProfile.name} -HTTP status code ${sdkResponse.httpStatusCode} - Failed to get company profile for company number ${companyNumber}`);
+    if (!sdkResponse.httpStatusCode || sdkResponse.httpStatusCode !== HttpStatusCode.Ok) {
+        throw createAndLogError(`${getCompanyProfile.name} -HTTP status code ${sdkResponse.httpStatusCode} - Failed to get company profile for company number ${companyNumber}`);
     }
 
-    if (!sdkResponse.resource) {
-        throw logger.info(`${getCompanyProfile.name} - Company Profile API returned no resource for company number ${companyNumber}`);
+    const castedSdkResponse = sdkResponse as Resource<CompanyProfile>;
```
Comment (hepsimo) [inline src/services/companyProfileService.ts:24]: Should guard against the cast failing here by checking sdkResponse is an ApiErrorResponse (has an "errors" property), and then exiting with createAndLogError, something like

```suggestion
    if ("errors" in sdkResponse) {
        throw createAndLogError(`${getCompanyProfile.name} - ${sdkResponse.errors} - Failed to get company profile for company number ${companyNumber}`);
    }

    const castedSdkResponse = sdkResponse as Resource<CompanyProfile>;
```

File: test/services/companyProfileService.unit.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        jest.clearAllMocks();
+    });
+
+    afterEach(() => {
+        expect(mockCreateOAuthApiClient).toHaveBeenCalledTimes(1);
+        expect(mockGetCompanyProfile).toHaveBeenCalledTimes(1);
+    });
+
+    describe("getPscVerification", () => {
+        it("should retrieve the company profile resource", async () => {
```
Comment (hepsimo) [inline test/services/companyProfileService.unit.ts:28]: Add a test scenario: getCompanyProfile() response is an ApiErrorResponse object with "errors" property.

**Review by hepsimo — CHANGES_REQUESTED** [review]
Changes look good. I suggested adding checks for ApiErrorResponse if you think it's useful/appropriate.

File: test/services/companyProfileService.unit.ts (hunk 2 above)
Comment (AlisonGriffiths) [reply test/services/companyProfileService.unit.ts:28]: I have added that additional scenario

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/services/companyProfileService.ts [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
-
-    logger.debug(`${getCompanyProfile.name} - Looking for company profile with company number ${companyNumber}`);
-    const sdkResponse: Resource<CompanyProfile> = await apiClient.companyProfile.getCompanyProfile(companyNumber);
+    logger.debug(`${getCompanyProfile.name} - Get company profile with company number ${companyNumber}`);
+    const sdkResponse: Resource<CompanyProfile> | ApiErrorResponse = await oAuthApiClient.companyProfile.getCompanyProfile(companyNumber);
 
     if (!sdkResponse) {
-        throw logger.info(`${getCompanyProfile.name} - Company Profile API returned no response for company number ${companyNumber}`);
+        throw createAndLogError(`${getCompanyProfile.name} - Company Profile API returned no response for company number ${companyNumber}`);
     }
```
Comment (hepsimo) [inline src/services/companyProfileService.ts:19]: We could provide more detail about the errors, if you think that's useful, something like:

```suggestion
    if (sdkResponse.httpStatusCode === HttpStatusCode.InternalServerError && "errors" in sdkResponse) {
        const messages = sdkResponse.errors?.map(e => e.error);
        throw createAndLogError(`${getCompanyProfile.name} - Company Profile API returned with errors for company number ${companyNumber}: ${messages}`);
    }
```

**Review by hepsimo — COMMENTED** [review]
Suggesion: more log detail on the ApiErrorResponse

**Review by hepsimo — APPROVED** [review]
Changes look good.

**Pull request merged**